src/components/ReduxFormCreatableSelect/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AsyncCreatableSelect from 'react-select/async-creatable';
+
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+
+const ReduxFormCreatableSelect = props => {
+  const {
+    input, defaultOptions, currValue, setValue,
+  } = props;
+
+  const loadOptions = (inputValue, callback) => DiscoveryDataApiService.fetchCourseTags(inputValue)
+    .then((response) => {
+      callback(response.data.map(tag => (
+        {
+          label: tag.value,
+          value: tag.value,
+        }
+      )));
+    })
+    .catch(() => {
+      callback(null);
+    });
+
+  return (
+
+    <>
+      {props.label}
+      <AsyncCreatableSelect
+        isMulti
+        onChange={value => {
+          input.onChange(value);
+          setValue(value ? value.map(v => v.label) : []);
+        }}
+        onBlur={() => input.onBlur(input.value)}
+        defaultOptions={defaultOptions}
+        isValidNewOption={props.createOptionValidator}
+        cacheOptions
+        isSearchable
+        styles={{
+          menu: styles => ({ ...styles, zIndex: 5 }),
+          control: styles => ({ ...styles, marginTop: '8px' }),
+        }}
+        value={currValue}
+        loadOptions={loadOptions}
+      />
+    </>
+  );
+};
+
+ReduxFormCreatableSelect.defaultProps = {
+
+};
```

Comment:
Is this necessary, considering there are no defaultProps?

Reply:
removed

Reply:
and added again since we have default props now